src/mococrw/asymmetric_crypto_ctx.h

```diff
@@ -0,0 +1,255 @@
+/*
+ * #%L
+ * %%
+ * Copyright (C) 2018 BMW Car IT GmbH
+ * %%
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+#pragma once
+
+#include "padding_mode.h"
+#include "x509.h"
+
+namespace mococrw {
+
+class AsymmetricCryptoCtx
+{
+public:
+
+    /**
+    * Forward declaration of nested Builder class
+    *
+    * Definition is below.
+    *
+    */
+    class Builder;
+
+    virtual ~AsymmetricCryptoCtx() = default;
+
+    virtual AsymmetricKey::KeyTypes getKeyType() const = 0;
+
+    /**
+     * Sets the RSA padding mode Object
+     * @param rsaPaddingMode padding mode to be used
+     */
+    void setRsaPaddingMode (std::shared_ptr<RSAPadding> const& rsaPaddingMode)
+    { _paddingMode = rsaPaddingMode; }
+
+    /**
+     * Gets the RSA padding mode object
+     * @return RSAPadding mode
+     */
+    std::shared_ptr<RSAPadding> getRsaPaddingMode() const
```

Comment:
see above comment on const correctness. While this is syntactically correct, I think this gives the wrong impression.

Reply:
Removed the method. It's unused and to be honest it doens't serve any specific purpose.